src/TrackGenerator.cpp

```diff
@@ -423,13 +406,27 @@ FP_PRECISION TrackGenerator::getMaxOpticalLength() {
           sigma_t = material->getSigmaT();
 
           for (int e=0; e < material->getNumEnergyGroups(); e++)
-            max_optical_length = std::max(max_optical_length, length*sigma_t[e]);
+            max_optical_length = std::max(max_optical_length,
```

Comment:
Not quite because we first need to compute the maximum optical length then use it for splitting segments for every ``MOCKernel`` created from then on. ``MOCKernels`` are temporary but the ``TrackGenerator`` endures for the entire calculation so the maximum optical path length is stored there.